tests/test_processing_common.py

```diff
@@ -82,6 +102,137 @@ def test_processor_from_and_save_pretrained(self):
 
                 self.assertEqual(processor_second.to_dict(), processor_first.to_dict())
 
+    # These kwargs-related tests ensure that processors are correctly instantiated.
+    # they need to be applied only if an image_processor exists.
+
+    def skip_processor_without_typed_kwargs(self, processor):
+        # TODO this signature check is to test only uniformized processors.
+        # Once all are updated, remove it.
+        is_kwargs_typed_dict = False
+        call_signature = inspect.signature(processor.__call__)
+        for param in call_signature.parameters.values():
+            if param.kind == param.VAR_KEYWORD and param.annotation != param.empty:
+                is_kwargs_typed_dict = (
+                    hasattr(param.annotation, "__origin__") and param.annotation.__origin__ == Unpack
+                )
```

Comment:
I'm assuming we don't `break` here as this is testing for kwargs, which will always be last?

Reply:
I'm sorry but I don't understand :grimacing:

Reply:
Sorry, I wasn't super clear. 

In the for-loop, iterating over the params either: 
1. Many params could satisfy `param.kind == param.VAR_KEYWORD and param.annotation != param.empty`, in which case `is_kwargs_typed_dict` can be set multiple times
2. Only one param could satisfy `param.kind == param.VAR_KEYWORD and param.annotation != param.empty`, so if we hit it, we could do an early exit

For both cases, my understanding is it's not an issues as `param.kind == param.VAR_KEYWORD` is effectively testing for `kwargs`, and so the condition is only satisfied once (resetting issue in 1. isn't realised) and they're always the final arguments, so there's no advantage from exiting early in 2.

Reply:
ah, got it, thanks for unrolling it :smile: 
I suppose you're right, I kind of assumed it was going to be always last, I think it's ok to leave it as is?

Reply:
Yep! I agree, I just wanted to make sure my understanding was correct :)

Reply:
dug a tad, couldn't find a way to break it so I think it is!